### PR TITLE
New committee detail

### LINKED ISSUE #403: /committee/id detail 1/2
make sql view

### LINKED ISSUE #406: /committee/id detail 2/2
write python of /committee/id (flask and sql alchemy)

### LINKED ISSUE #435: /candidate/<id> endpoint
We have all the data we are using on the `/candidates` endpoint. To get rid of all of the old stuff we should replace this functionality so you can find more detailed information like candidate mailing addresses and committees affiliated with a candidate on an endpoint.

### PR DESCRIPTION
**Pull request by LindsayYoung**
This gets the following endpoints working:
 - `/committee/<committee_id>` This shows all the committee details for one committee.
 - `/candidate/<candidate_id>/committees` This shows all the committee details for all committees related to one candidate.
 - `/candidate/<candidate_id>` This shows all the candidate details for one candidate.
 - `/committee/<committee_id>/candidates` This shows all the candidate details for all the candidates related to one committee.

We still need to:
-  Make the information on the detail views change with year, so if a year is supplied, the right ici_status is shown
-  Add Candidate's mailing address to the candidate detail view- I couldn't get that to link

fix #435 
fix #406 
fix #403

@theresaanna @noahmanger  This will break anything that is still using the old `/candidate/<id.>` and `/committee/<id>` endpoints. Mostly because the structure is flattened.

### CODE REVIEW

**Comment by LindsayYoung** [comment]
There is a typo in the travis file I will fix now.

**Comment by LindsayYoung** [comment]
I found a pagination error and committees are showing up multiple times. I will fix that now.

**Comment by rjmajma** [comment]
Travis, you can be a real jerk sometimes.

**Comment by LindsayYoung** [comment]
This is ready for a merge. 

We still need to:
 - make the information on the detail views change with year, so if a year is supplied, the right ici_status is shown.  @theresaanna because that is no longer responsive we are not ready to implement year filtering on the candidate page.

- only show the most recent linking of a candidate in the committee detail and the most recent committee in the candidate detail.

- Fix address. We are not using it on the website, so I don't think we should block this PR but `ofec_candidate_detail_vw` is still messed up. The join on dimcandproperties is 

### Changes

 - don't use `fields=*`

On the `/candidate/<candidate_id>` and `/committee/<committee_id>/candidates` endpoints:

 - party_affilation => party_full
 - office_sought => office
 - election_year => active_though
 - Incumbent_challange_full now has a longer description because I am getting it from a different table. Now it says Incumbent (Current seat holder running for election) @noahmanger is this helpful or just too long? I can change it back if you want. 

The address fields will be flattened and their names will have `address_` in front of them. The primary purpose of that is so people don't get confused and use the wrong state because the mailing address of the candidate or committee can be different from the state the candidate is running for office.

- The old endpoint can be accessed at `/oldcandidate` I don't think we should get rid of it until the detailed committee information is year sensitive

Here is the diagram of the flow:
img_0760

File: data/sql_updates/create_candidate_detail_vw.sql [hunk 1]
```diff
@@ -0,0 +1,56 @@
+drop view if exists ofec_candidate_detail_vw;
+create view ofec_candidate_detail_vw as
+select
+    max(dimcand.cand_sk) as candidate_key,
```
Comment (arowla) [inline data/sql_updates/create_candidate_detail_vw.sql]: Anything in the GROUP BY should not be using an aggregate function. Remove `max()`.

File: data/sql_updates/create_candidate_detail_vw.sql [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    ) csi_recent using (cand_sk)
+    left join dimcandstatusici csi_all using (cand_sk)
+    left join dimcandoffice co on co.cand_sk = dimcand.cand_sk and (csi_recent.election_yr is null or co.cand_election_yr = csi_recent.election_yr)  -- only joined to get to dimoffice
+    inner join dimoffice using (office_sk)
+    inner join dimparty using (party_sk)
+    left join (
+        select distinct on (cand_sk) * from dimcandproperties order by cand_sk desc limit 1
+    ) cand_p_most_recent on cand_p_most_recent.cand_sk = cand_p_most_recent.cand_sk
+group by
+    dimcand.cand_sk
```
Comment (arowla) [inline data/sql_updates/create_candidate_detail_vw.sql]: Shouldn't `cand_id` also be in the GROUP BY?

**Pull request merged**